searchcore/src/vespa/searchcore/proton/persistenceengine/transport_latch.cpp

```diff
@@ -8,10 +8,50 @@ using storage::spi::Result;
 
 namespace proton {
 
+std::unique_ptr<std::mutex> createOptionalLock(bool needLocking) {
+    return needLocking
+        ? std::make_unique<std::mutex>()
+        : std::unique_ptr<std::mutex>();
+}
+TransportMerger::TransportMerger(bool needLocking)
+    : _result(),
+      _lock(createOptionalLock(needLocking))
+{
+}
+TransportMerger::~TransportMerger() = default;
+
+void
+TransportMerger::mergeResult(ResultUP result, bool documentWasFound) {
+    if (_lock) {
+        std::lock_guard<std::mutex> guard(*_lock);
+        mergeWithLock(std::move(result), documentWasFound);
+    } else {
+        mergeWithLock(std::move(result), documentWasFound);
+    }
+}
+
+void
+TransportMerger::mergeWithLock(ResultUP result, bool documentWasFound) {
```

Comment:
Consider adding a reference to the lock to ensure that we don't call this when the lock is not taken.

Reply:
I did not do that sine the lock is optional.